app/apollo/resolvers/common.js

```diff
@@ -55,53 +55,76 @@ const validClusterAuth = async (me, queryName, context) => {
   }
 };
 
-var getAllowedChannels = async(me, org_id, action, field, context)=>{
+// Polymorphically check for cached IAM decision, Get resources authorized by Access Policy, Update cache for individual resource authentication
+var getAllowedResources = async(me, org_id, action, type, queryName, context, searchByTags=null, searchByUuid=null, searchQuery=null)=>{
   const { models } = context;
-  var channels = await models.Channel.find({ org_id });
-  return await filterChannelsToAllowed(me, org_id, action, field, channels, context);
-};
 
-var filterChannelsToAllowed = async(me, org_id, action, field, channels, context)=>{
-  const { models } = context;
-  var decisionInputs = _.map(channels, (channel)=>{
-    return {
-      type: field,
-      action,
-      uuid: channel.uuid,
-      name: channel.name,
-    };
-  });
-  var decisions = await models.User.isAuthorizedBatch(me, org_id, decisionInputs, context);
-  channels = _.filter(channels, (val, idx)=>{
-    return decisions[idx];
-  });
-  return channels;
-};
+  var resources;
 
-var getAllowedSubscriptions = async(me, org_id, action, field, context)=>{
-  const { models } = context;
-  var subscriptions = await models.Subscription.find({ org_id });
-  return await filterSubscriptionsToAllowed(me, org_id, action, field, subscriptions, context);
+  // Check for cached IAM decision, returns true if cache is empty or all is authorized
```

Comment:
Function is incorrectly described (or if accurate, incorrectly used).  E.g. if function returns `true` due to cache miss, this function will incorrectly treat that identically to "user **does** have auth to all resources of this type".

Reply:
The comment is what was wrong here. I updated the comments to reflect behavior. Check for cached IAM decision. Returns true if all is authorized under passed resource type, false if not all authorized or cache is empty. If false, use fine grained auth